site/en/blog/framework-tools-font-fallback/index.md

```diff
@@ -0,0 +1,390 @@
+
+
+## Background
+
+_[font-display: swap](/blog/font-display/#swap)_ is generally used to prevent FOIT (Flash of invisible text) and to display contents faster on the screen. The value of `swap` tells the browser that text using the font should be displayed immediately using a system font and to replace the system font only when the custom font is ready.
+
+The biggest issue with `swap` is the jarring effect, where the difference in character sizes of the two fonts results in screen content shifting around. This leads to poor CLS scores, especially for text-heavy websites. 
+
+The following images show an example of the issue. The first image uses `font-display: swap` with no attempt to adjust the size of the fallback font. The second shows how adjusting the size using the CSS `@font-face` rule improves the loading experience.
+
+
+### Without adjusting font size
+
+```css
+body {
+  font-family: Inter, serif;
+}
+```
+
+<figure>
+{% Img src="image/IypihH3o5cSpEMVp5i08dp69otp2/AZ91M4LW9H8DzgOenxDB.gif", alt="Text that suddenly changes in font and size causing a jarring effect.", width="359", height="450" %}
+</figure>
+
+
+
+### After adjusting font size
+
+```css
+body {
+  font-family: Inter, fallback-inter, serif;
+  }
+
+@font-face {
+  font-family: "fallback-inter";
+  ascent-override: 90.20%;
+  descent-override: 22.48%;
+  line-gap-override: 0.00%;
+  size-adjust: 107.40%;
+  src: local("Arial");
+}
+```
+
+<figure>
+{% Img src="image/IypihH3o5cSpEMVp5i08dp69otp2/AjGJB22nMK3WtqCIx7V0.gif", alt="Text that transitions smoothly to a different font.", width="359", height="450" %}
+</figure>
+
+
+Adjusting the size of the fallback font can be an effective strategy for preventing font loading layout shift, but implementing the logic from scratch can be tricky, as described in [this post about font fallbacks](/blog/font-fallbacks/). Fortunately, several tooling options are already available to make this easier while developing apps. 
+
+
+## How to optimize font fallbacks with Next.js
+
+Next.js provides a built-in way to enable fallback font optimization. This feature is enabled by default when you load fonts using the [@next/font](https://nextjs.org/docs/api-reference/next/font) component.
+
+The [@next/font](https://nextjs.org/docs/api-reference/next/font) component was introduced in Next.js version 13. The component provides an API to import Google Fonts or custom fonts into your pages and includes built-in automatic self-hosting of font files. 
+
+When used, the fallback font metrics are automatically calculated and injected into the CSS file.
+
+​​For example, if you are using a Roboto font, you would typically define it in CSS as follows:
+
+```css
+@font-face {
+  font-family: 'Roboto';
+  font-display: swap;
+  src: url('/fonts/Roboto.woff2') format('woff2'), url('/fonts/Roboto.woff') format('woff');
+  font-weight: 700;
+}
+
+body {
+  font-family: Roboto;
+}
+
```

Comment:
This code snippet is missing the "end" tick marks (```), so the next section is in the code snippet

Reply:
Updated.